### DISCUSSION TITLE
GUI Titlebar & General Styling Support

### DISCUSSION THREAD

**Discussion by nytokio** [AllenDang/giu#900] [General]
Hello 👋

This is my first time using GIU, I am having a few issues figuring some stuff out. I'll list below some items I could use help figure out.

Styling the title bar: 
![image](https://github.com/user-attachments/assets/1caf273e-353f-44c6-8602-292cfc5c045f)

setting the MasterWindow Backgorund, and making buttons, dropdowns, and other items look more like Dear ImGUI: 
![image](https://github.com/user-attachments/assets/29113377-23f7-4075-a693-077772aef631)

Furthermore, the default hover color of buttons and I think items are light blue? Any way of changing this?

![image](https://github.com/user-attachments/assets/bed6d69c-1bd7-4437-b197-0e397136636d)

**Comment by SleepyPrince** [comment]
You can use css [doc](https://github.com/AllenDang/giu/blob/master/docs/css.md) [example](https://github.com/AllenDang/giu/tree/master/examples/CSS-styling)
Not sure about the title bar though

**Comment by gucio321** [comment]
Hi @Kiritsuu nice to hear that someone uses giu :smile:

> Styling the title bar:

unfortunately this is something handled by glfw and it depends on platform (e.g. for me on linux this title bar is different)
![image](https://github.com/user-attachments/assets/dd9566ee-dc42-42c2-86f7-f8df5ae5185f)
You can however disable it: `giu.MasterWindowFlagsFrameless`

> setting the MasterWindow Backgorund, and making buttons, dropdowns, and other items look more like Dear ImGUI:

generally this is set by `setTheme` function (there is an issue #773 that would allow to overwrite this). For now you need to use StyleSetter API
```go
giu.Style().
   SetColor(colid1, colornames.Red).
   SetColor(colid2, colornames.Blue).
   SetStyle(styleid1, 20, 40).
   SetStyleFloat(styleid2, 8.5) // e.t.c.
```

Now depending on what you want to do you can either use Push/Pop or `(*StyleSetter).To` method.

```go
giu.SingleWindow().Layout(
   giu.Style()./*your style settings like above*/.To(
      widget1,
      widget2,
   ),
)
```
or for the whole app you can do like this:
```go
func loop() {
   myStyle := Style()./*your style settings*/
   myStyle.Push()
   // render your app (SingleWindow, widgets e.t.c.)
   myStyle.Pop()
}
```
`StyleSetter` usage is presented in [setstyle example](https://github.com/AllenDang/giu/tree/master/examples/setstyle)

You can also use CSS styling if you prefer :smile:
see [this example](https://github.com/AllenDang/giu/tree/master/examples/CSS-styling)

here is also something that you might want to check: https://github.com/AllenDang/giu/wiki#style-related
